api/helpers/tell-me/contributorParsing.js

```diff
@@ -0,0 +1,168 @@
+ */
+
+/**
+ * @typedef FormattedContributor
+ * @type {Omit<
+ *  import("@prisma/client").Contributor,
+ * 'createdAt' | 'updatedAt' | 'id' | 'category' | 'position' | 'contactCategoryId'
+ * >}
+ */
+
+/** @type {"NONE"|"RAW"|"CONSOLIDATED"} */
+const CONTRIBUTOR_SUBMISSION_PARSING_MODE = 'CONSOLIDATED'
+
+const FIELD_MAP = {
+  // TODO: add CONTRIBUTOR_TYPE handling
+  email: 'EMAIL',
+  firstName: 'FIRSTNAME',
+  lastName: 'LASTNAME',
+  phone: 'PHONE',
+}
+
+class TellMeContributorSubmission {
+  /**
+   * @param {object} submission
+   */
+  constructor(submission) {
+    this.rawSubmission = submission
+    this.submissionId = this.getSubmissionId()
+    this.consolidatedKeys = []
+  }
+
+  /**
+   * @param {string} fieldName
+   * @param {FormattedSubmission} submission
+   * @returns {string}
+   */
+  getFieldValue(fieldName, submission) {
+    switch (CONTRIBUTOR_SUBMISSION_PARSING_MODE) {
+      case 'CONSOLIDATED':
+        return this.getMappedFieldValue(fieldName, submission)
+      case 'RAW':
+      case 'NONE':
+      default:
+        switch (fieldName) {
+          case 'firstName':
+            return submission.id
+          default:
+            return '...'
+        }
+    }
+  }
+
+  /**
+   * @param {string} fieldName
+   * @param {FormattedSubmission} submission
+   * @returns {string}
+   */
+  getMappedFieldValue(fieldName, submission) {
+    const mappedKey = FIELD_MAP[fieldName]
+    const mappedAnswer = submission.answers.find(answer => mappedKey === answer.key)
+    if (mappedAnswer === undefined) {
+      return '...'
+    }
+    this.consolidatedKeys.push(mappedKey)
+
+    return mappedAnswer.value
+  }
+
+  /**
+   * @returns {Array<FormattedAnswer>}
+   */
+  extractAnswers() {
+    return this.rawSubmission.answers.map(
+      /** @returns {FormattedAnswer} */
+      answer => ({
+        id: answer.question.id,
+        key: answer.question.key,
+        label: answer.question.value,
+        value: answer.rawValue,
+      }),
+    )
+  }
+
+  getSubmissionId() {
+    return `${this.rawSubmission.openedAt}|${this.rawSubmission.submittedAt}`
+  }
+
+  /**
+   * @returns {FormattedSubmission}
+   */
+  extractSubmission() {
+    return {
+      answers: this.extractAnswers(),
+      id: this.submissionId,
+      openedAt: this.rawSubmission.openedAt,
+      submittedAt: this.rawSubmission.submittedAt,
+    }
+  }
+
+  /**
+   * @param {FormattedAnswer} answer
+   * @returns {string}
+   */
+  // eslint-disable-next-line class-methods-use-this
+  formatNoteAnswer(answer) {
+    // eslint-disable-next-line no-irregular-whitespace
+    return `${answer.label.trim()} ${answer.value.trim()}`
+  }
+
+  /**
+   * @param {FormattedSubmission} submission
+   * @returns {string}
+   */
+  formatSubmissionForNotes(submission) {
+    let notes
+    switch (CONTRIBUTOR_SUBMISSION_PARSING_MODE) {
+      case 'RAW':
+        notes = submission.answers.map(answer => this.formatNoteAnswer(answer))
+        break
```

Comment:
I'm surprised the linter is ok with that. I will make a pass on inca-app to update the rules I think.

A `break` should always be preceded by an empty line unless after a condition line.

And `case` / `default` should be preceded by an empty line unless after a switch line.

Reply:
Ok
(I admit I don't like so much your empty line config ;) Particularly the return one)